sdk/src/test/java/com/mapbox/maps/renderer/MapboxRenderThreadTest.kt

```diff
@@ -795,4 +796,53 @@ class MapboxRenderThreadTest {
     verify { runnable.run() }
     verify { runnable2.run() }
   }
+
+  @Test
+  fun contextLostTest() {
+    initRenderThread()
+    provideValidSurface()
+    pauseHandler()
+    every { eglCore.swapBuffers(any()) } returns EGL11.EGL_CONTEXT_LOST
+    mapboxRenderThread.queueRenderEvent(MapboxRenderer.repaintRenderEvent)
+    idleHandler()
+    every { eglCore.swapBuffers(any()) } returns EGL11.EGL_SUCCESS
+    mapboxRenderThread.queueRenderEvent(MapboxRenderer.repaintRenderEvent)
+    idleHandler()
+    // one swap buffer for surface creation, one for EGL_CONTEXT_LOST, one for EGL_SUCCESS
+    verify(exactly = 3) {
+      eglCore.swapBuffers(any())
+    }
+    // EGL and native renderer are recreated
+    verifyOrder {
```

Comment:
Check that recreation happens right after swap buffers EGL_CONTEXT_LOST?

Reply:
done